### DISCUSSION TITLE
How to plot network architecture?

### DISCUSSION THREAD

**Discussion by rmaria** [apache/incubator-mxnet#10269] [Q&A]
Note: Providing complete information in the most concise form is the best way to get help. This issue template serves as the checklist for essential information to most of the technical issues and bug reports. For non-technical issues and feature requests, feel free to present the information in what you believe is the best form.

For Q & A and discussion, please start a discussion thread at https://discuss.mxnet.io 

## Description
I have a pre-trained model saved in the format: name.params, name.prototx, name-symbol.json
I am loading the model using net = mx.model.FeedForward.load(name, 1)
I want to visualize the model using mx.viz.plot_network(net), but I get the error: "symbol must be a Symbol"

## Environment info (Required)
python virtual environment
Ubuntu 16.04
mxnet 1.2.0
graphviz installed
```
What to do:
1. Download the diagnosis script from https://raw.githubusercontent.com/apache/incubator-mxnet/master/tools/diagnose.py
2. Run the script using `python diagnose.py` and paste its output here.

Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                8
On-line CPU(s) list:   0-7
Thread(s) per core:    2
Core(s) per socket:    4
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 158
Model name:            Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
Stepping:              9
CPU MHz:               2800.000
CPU max MHz:           3800,0000
CPU min MHz:           800,0000
BogoMIPS:              5616.00
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              6144K
NUMA node0 CPU(s):     0-7
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single pti retpoline intel_pt rsb_ctxsw tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx rdseed adx smap clflushopt xsaveopt xsavec xgetbv1 xsaves dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp
----------Python Info----------
Version      : 3.5.2
Compiler     : GCC 5.4.0 20160609
Build        : ('default', 'Nov 23 2017 16:37:01')
Arch         : ('64bit', 'ELF')
------------Pip Info-----------
Version      : 9.0.2
Directory    : /home/merlin/.virtualenvs/dl4cv/lib/python3.5/site-packages/pip
----------MXNet Info-----------
Version      : 1.2.0
Directory    : /home/merlin/.virtualenvs/dl4cv/lib/python3.5/site-packages/mxnet
Hashtag not found. Not installed from pre-built package.
----------System Info----------
Platform     : Linux-4.13.0-36-generic-x86_64-with-Ubuntu-16.04-xenial
system       : Linux
node         : merlin-pc
release      : 4.13.0-36-generic
version      : #40~16.04.1-Ubuntu SMP Fri Feb 16 23:25:58 UTC 2018
----------Hardware Info----------
machine      : x86_64
processor    : x86_64
----------Network Test----------
Setting timeout: 10
Timing for MXNet: https://github.com/apache/incubator-mxnet, DNS: 0.0062 sec, LOAD: 1.2192 sec.
Timing for PYPI: https://pypi.python.org/pypi/pip, DNS: 0.0367 sec, LOAD: 0.7809 sec.
Timing for FashionMNIST: https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/fashion-mnist/train-labels-idx1-ubyte.gz, DNS: 0.0770 sec, LOAD: 0.9621 sec.
Timing for Conda: https://repo.continuum.io/pkgs/free/, DNS: 0.0469 sec, LOAD: 0.6517 sec.
Timing for Gluon Tutorial(en): http://gluon.mxnet.io, DNS: 0.0991 sec, LOAD: 0.1915 sec.
Timing for Gluon Tutorial(cn): https://zh.gluon.ai, DNS: 0.1499 sec, LOAD: 1.0153 sec.

```

Package used (Python/R/Scala/Julia):
(I'm using ...)

For Scala user, please provide:
1. Java version: (`java -version`)
2. Maven version: (`mvn -version`)
3. Scala runtime if applicable: (`scala -version`)

For R user, please provide R `sessionInfo()`:

## Build info (Required if built from source)

Compiler (gcc/clang/mingw/visual studio):

MXNet commit hash:
(Paste the output of `git rev-parse HEAD` here.)

Build config:
(Paste the content of config.mk, or the build command.)

## Error Message:
(Paste the complete error message, including stack trace.)

## Minimum reproducible example
(If you are using your own code, please provide a short script that reproduces the error. Otherwise, please provide link to the existing example.)

## Steps to reproduce
(Paste the commands you ran that produced the error.)

1.net = mx.model.FeedForward.load(model,1)
2.mx.viz.plot_network(net)

## What have you tried to solve it?

1. to visualize the architecture of a pre-trained network
2.

**Comment by reminisce** [comment]
Try this `mx.viz.plot_network(net.symbol)`.
BTW, we are going to deliver a more powerful solution using TensorBoard for visualizing networks and other data types of MXNet. Stay tuned.

**Comment by ShownX** [comment]
I am waiting for this feature for more than half a year :) I am so glad it is on the way.

**Comment by reminisce** [comment]
@rmaria @ShownX We have open-sourced [MXBoard](https://github.com/awslabs/mxboard) which enables users to visualize MXNet data (including network structures) in [TensorBoard](https://pypi.python.org/pypi/tensorboard). The current version on PyPI is [0.1.0rc3](https://pypi.python.org/pypi?:action=display&name=mxboard&version=0.1.0rc3). You can either install from source or `pip install`. The GitHub site has rich examples and tutorials on installation and how to use it in MXNet. Could you try it and give us feedbacks? Thanks.

**Comment by ShownX** [comment]
Sure. Thank you for this tool.

**Comment by ThomasDelteil** [comment]
@rmaria  please consider closing if your issue is solved :+1:

**Comment by safrooze** [comment]
@rmaria I believe your question is now answered. Please close this issue.
